SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -7363,14 +7363,23 @@ void CEulerSolver::BC_Supersonic_Inlet(CGeometry *geometry, CSolver **solver_con
    so all flow variables can be imposed at the inlet.
    First, retrieve the specified values for the primitive variables. ---*/
 
-  const su2double Temperature = config->GetInlet_Temperature(Marker_Tag) / config->GetTemperature_Ref();
-  const su2double Pressure = config->GetInlet_Pressure(Marker_Tag) / config->GetPressure_Ref();
-  const auto* Vel = config->GetInlet_Velocity(Marker_Tag);
+  /*--- Loop over all the vertices on this boundary marker ---*/
 
-  su2double Velocity[MAXNDIM] = {0.0};
-  for (unsigned short iDim = 0; iDim < nDim; iDim++)
-    Velocity[iDim] = Vel[iDim] / config->GetVelocity_Ref();
+  SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+  for (auto iVertex = 0ul; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
 
+    if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+    /*--- Retrieve the specified conditions for this inlet. ---*/
+
+    const su2double Temperature = Inlet_Temperature[val_marker][iVertex] / config->GetTemperature_Ref(); 
+    const su2double Pressure = Inlet_Pressure[val_marker][iVertex] / config->GetPressure_Ref(); 
+    const auto* Vel = Inlet_Velocity[val_marker][iVertex];
+
+    su2double Velocity[MAXNDIM] = {0.0};
+    for (unsigned short iDim = 0; iDim < nDim; iDim++)
+      Velocity[iDim] = Vel[iDim] / config->GetVelocity_Ref(); 
```

Comment:
You need to make these changes in this section around line 7400
```
 V_inlet[prim_idx.Temperature()] = Temperature;
    V_inlet[prim_idx.Pressure()] = Pressure;
    V_inlet[prim_idx.Density()] = Density;
    V_inlet[prim_idx.Enthalpy()] = Energy + Pressure / Density;
    for (unsigned short iDim = 0; iDim < nDim; iDim++)
      V_inlet[iDim+prim_idx.Velocity()] = Velocity[iDim];
``` 
Replacing "Temperature" with `Inlet_Ttotal[val_marker][iVertex] / config->GetTemperature_Ref()` etc.

Reply:
Okay. Just to make sure that is the idea currently using the name of 'total'(so that needn't define new variables) but still actually putting the static values at each vertex as the original supersonic inlet do?

Reply:
That is correct. We already "overload" the meaning of "Ptotal" for mass flow inlets and velocity inlets (in incompressible flow).

Reply:
Hi, see if codes around 7400 are correct now. And I will recompile and test tomorrow morning in Beijing time. Thanks for the help!